packages/framework-provider-aws/src/library/read-models-adapter.ts

```diff
@@ -69,7 +69,21 @@ export async function fetchReadModel(
     `[ReadModelAdapter#fetchReadModel] Loaded read model ${readModelName} with ID ${readModelID} with result:`,
     response.Items
   )
-  return response.Items as unknown as ReadOnlyNonEmptyArray<ReadModelInterface>
+
+  if (!response.Items || response.Items?.length === 0) {
+    return response.Items as unknown as ReadOnlyNonEmptyArray<ReadModelInterface>
+  }
+
+  return (response.Items as unknown as ReadOnlyNonEmptyArray<ReadModelInterface>).map((value) => {
+    const optimisticConcurrencyValue = generateOptimisticConcurrencyValue(value)
+    return {
+      ...value,
+      boosterMetadata: {
+        ...value?.boosterMetadata,
+        optimisticConcurrencyValue: optimisticConcurrencyValue,
```

Comment:
I guess you can use this syntax here:

```suggestion
        optimisticConcurrencyValue,
```

Reply:
Awesome, I didn't know it